internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -699,101 +696,105 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				ForceNew: true,
 			},
 
-			"role_based_access_control": {
+			"role_based_access_control_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
```

Comment:
this'd be a breaking change right now, should this be feature-flagged behind 3.0?